modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+<?php
+
+/**
+ * @file
+ * Handle the integration of digital article service (DAS) into the CMS.
+ */
+
+/**
+ * Implements hook_das_menu().
+ */
+function ting_das_menu() {
+  $items = array();
+
+  $items['admin/config/ting/das'] = array(
+    'title' => 'Digital article service',
+    'description' => 'Configure access to DAS.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ting_das_admin_settings_form'),
+    'access arguments' => array('administer site configuration'),
+    'file' => 'includes/ting_das.admin.inc',
+  );
+
+  $items['das/%ting_object/order'] = array(
+    'page callback' => 'ting_das_ajax',
+    'page arguments' => array(1),
+    'delivery callback' => 'ajax_deliver',
+    'access arguments' => array('access content'),
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_libraries_info().
+ *
+ * For defining external libraries.
+ */
+function ting_das_libraries_info() {
+  return array(
+    'guzzle' => array(
+      'name' => 'Guzzle',
+      'vendor url' => 'https://github.com/guzzle/guzzle',
+      'download url' => 'https://github.com/guzzle/guzzle',
+      'version' => '6.2',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+  );
+}
+
+/**
+ * Implements hook_ding_entity_buttons().
+ */
+function ting_das_ding_entity_buttons($type, $entity, $view_mode, $widget = 'default') {
+  $type = drupal_strtolower($entity->getType());
+  $types = array_filter(_ting_das_get_default_well_types());
+
+  // Check that the entity is not already available as an online resources and is one of the configured types.
+  if (isset($types[$type]) && !$entity->is('online') && !$entity->is('infomedia')) {
+    drupal_add_library('system', 'drupal.ajax');
+
+    $display_button = TRUE;
+    if (user_is_logged_in()) {
+      try {
+        $creds = ding_user_get_creds();
+        $display_button = isset($creds['resident']) && $creds['resident'];
+      }
+      catch (DingProviderAuthException $exception) {
+        // This is okay as the use simply was not logged in as an provider
+        // user. So do noting in this case.
+      }
+    }
+
+    if ($display_button) {
+      return array(
+        array(
+          '#theme' => 'link',
+          '#text' => t('Order digital copy'),
+          '#path' => 'das/' . $entity->id . '/order',
+          '#options' => array(
+            'attributes' => array(
+              'class' => array(
+                'action-button',
+                'button-see-online',
+                'use-ajax',
+              ),
+              'id' => 'das-order-' . $entity->id,
+              'onclick' => 'wt.sendinfo({ linkId: "open_order_digital_copy" });',
+            ),
+            'html' => FALSE,
+          ),
+          '#attached' => array(
+            'css' => array(
+              drupal_get_path('module', 'ting_das') . '/css/ting_das.css',
```

Comment:
As far as I can tell the content in `ting_das` does not style the button at all. Perhaps it should be returned as a part of the AJAX response?

Reply:
It styles all content inside the "popup-bar" that drops down when the button is pressed. As I understands it, it's not possible to load it in an drupal ajax request as it will not add css to the header dynamically.

The "ajax_command_css($selector, $argument)" is only for adding inline css and I don't think that we should move the style from file into the module code to x linjes of code.